walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -15,9 +15,126 @@
 package com.google.sps;
 
 import java.util.Collection;
+import java.util.Collections;
+import java.util.Iterator;
+import java.util.Arrays;
+import java.util.ArrayList; // import the ArrayList class
+import java.util.Set;
+
+
+public final class FindMeetingQuery {    
+  ArrayList<TimeRange> options;
+
+  /*Returns true if the time slot is a possible option with respect to duration*/
+  public boolean slotPossibleDuration(MeetingRequest request, int startSlot, int endSlot){
+      return (request.getDuration() <= (endSlot - startSlot));
+  }
+
+  public boolean conflictIsRelevant(Event conflict, Collection meetingAttendees){
+    Set<String> conflictAttendees = conflict.getAttendees();
+    // Collection<String> meetingAttendees = request.getAttendees();
+
+    // returns true if the two sets have at least one attendee in common
+    return !(Collections.disjoint(conflictAttendees, meetingAttendees));
+  }
+
+  public boolean conflictIsOptionallyRelevant(Event conflict, MeetingRequest request){
+    Set<String> conflictAttendees = conflict.getAttendees();
+    Collection<String> optionalAttendees = request.getOptionalAttendees();
+
+    // returns true if the two sets have at least one optional attendee in common
+    return !(Collections.disjoint(conflictAttendees, optionalAttendees));  }
 
-public final class FindMeetingQuery {
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    // Create collection of timeranges representing options for meeting
+    Collection<TimeRange> optionsCollection;
```

Comment:
Variable name should mention something to do with timeRanges or availability. `options` is a bit ambiguous